docs/machine-learning/how-to-guides/inspect-intermediate-data-ml-net.md

```diff
@@ -1,75 +1,151 @@
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 200000f, 250000f, 230000f },
+        CurrentPrice = 225000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 126000f, 130000f, 200000f },
+        CurrentPrice = 195000f
+    },
+    new HousingData
+    {
+        Size = 850f,
+        HistoricalPrices = new float[] { 150000f,175000f,210000f },
+        CurrentPrice = 205000f
+    },
+    new HousingData
+    {
+        Size = 900f,
+        HistoricalPrices = new float[] { 155000f, 190000f, 220000f },
+        CurrentPrice = 210000f
+    },
+    new HousingData
+    {
+        Size = 550f,
+        HistoricalPrices = new float[] { 99000f, 98000f, 130000f },
+        CurrentPrice = 180000f
+    }
+};
+```
 
-During the experiment, you may want to observe and validate the data processing results at a given point. This isn't easy since ML.NET operations are lazy, constructing objects that are 'promises' of data.
+## Convert IDataView to IEnumerable
 
-The `GetColumn<T>` extension method lets you inspect the intermediate data. It returns the contents of one data column as an `IEnumerable`.
+One of the simplest ways to inspect the values of an IDataView is to convert it to an [`IEnumerable`](xref:System.Collections.Generic.IEnumerable`1). To convert an [`IDataView`](xref:Microsoft.ML.IDataView) to [`IEnumerable`](xref:System.Collections.Generic.IEnumerable`1) use the [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) method.
 
-The following example shows how to use the `GetColumn<T>` extension method:
+> [!WARNING]
+> [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) will load the entire [`IDataView`](xref:Microsoft.ML.IDataView) into memory which may affect performance.
 
-[Example file](https://github.com/dotnet/machinelearning/tree/master/test/data/adult.tiny.with-schema.txt):
+```csharp
+// Create an IEnumerable from IDataView
+HousingData[] housingDataEnumerable =
+    mlContext.Data.CreateEnumerable<HousingData>(data, reuseRowObject: false).ToArray();
 
-<!-- markdownlint-disable MD010 -->
+// Get first element of IEnumerable
+HousingData firstRow = housingDataEnumerable[0];
 ```
-Label	Workclass	education	marital-status
-0	Private	11th	Never-married
-0	Private	HS-grad	Married-civ-spouse
-1	Local-gov	Assoc-acdm	Married-civ-spouse
-1	Private	Some-college	Married-civ-spouse
 
+## Inspect values in a single column
+
+At any point in the process, values in a single column of an [`IDataView`](xref:Microsoft.ML.IDataView) can be accessed using the [`GetColumn`](xref:Microsoft.ML.Data.ColumnCursorExtensions.GetColumn*) method. The [`GetColumn`](xref:Microsoft.ML.Data.ColumnCursorExtensions.GetColumn*) method returns all of the values in a single column as an [`IEnumerable`](xref:System.Collections.Generic.IEnumerable`1)
+
+```csharp
+IEnumerable<float> sizeColumn = data.GetColumn<float>("Size").ToList();
```

Comment:
Is the syntax easy to work out if you have a vector column?

Reply:
Yep. For vectors you can use built-in types. It would be `float[]` when extracting (float) vectors.

Reply:
#resolved